pkg/controller/daemon/update.go

```diff
@@ -399,7 +399,7 @@ func (dsc *DaemonSetsController) controlledHistories(ds *apps.DaemonSet) ([]*app
 
 	// List all histories to include those that don't match the selector anymore
 	// but have a ControllerRef pointing to the controller.
-	histories, err := dsc.historyLister.List(labels.Everything())
+	histories, err := dsc.historyLister.ControllerRevisions(ds.Namespace).List(labels.Everything())
```

Comment:
this looks like in the right direction(based on the description in the bug)

Reply:
yes，controllerrevision is a namespaced resource

Reply:
this looks correct. can we also add safeguards into the owner/ownerref controller helpers (e.g. ClaimControllerRevisions, NewControllerRevisionControllerRefManager) to make adding cross-namespace owner references impossible?

Reply:
> this looks correct. can we also add safeguards into the owner/ownerref controller helpers (e.g. ClaimControllerRevisions, NewControllerRevisionControllerRefManager) to make adding cross-namespace owner references impossible?

good suggestion. add namespace check in BaseControllerRefManager.ClaimObject, please review again